### DISCUSSION TITLE
More than one MCP230XX on ESP8266

### DISCUSSION THREAD

**Discussion by schluo** [arendst/Tasmota#20736] [Q&A]
Dear Tasmota Community,

Two questions regarding the topic above which costs me a couple of hours so far :(

1. Right now I am using two MCP23017 extender in a Mode 2 config to control 32 relais. All 32 ports are configured per pseudo rule template as iRelay (256..287). Relais 1-31 are working as expected. Relais 32 is always activated. I have no idea why GPIO32 is always on high.

2. Is there a way to add more than 2 MCPs to the I2C bus (the template which defines the function of the GPIOs seems to supports only 32 different GPIOs)

Any idea, support, hint is very appreciated 🙂
Best regards,
Oliver

**Comment by barbudor** [comment]
1. Not sure. There might be a limitation to 31 relay on esp32
2. You can have up to 6 MCP
which doesn't change the limits on each type of gpio : relay, button, switch
So don't expect to get more than 32(31?) relay output
